numba/core/typing/new_builtins.py

```diff
@@ -982,17 +980,13 @@ def generic(self, args, kws):
 
         [arg] = args
 
-        if isinstance(arg, types.Integer):
-            return signature(arg, arg)
-        if isinstance(arg, (types.Float, types.Boolean)):
-            return signature(types.np_intp, arg)
-        if isinstance(arg, types.NPDatetime):
-            if arg.unit == 'ns':
-                return signature(types.np_int64, arg)
-            else:
-                raise errors.NumbaTypeError(f"Only datetime64[ns] can be converted, but got datetime64[{arg.unit}]")
-        if isinstance(arg, types.NPTimedelta):
-            return signature(types.np_int64, arg)
+        if isinstance(arg, types.NPDatetime) and arg.unit != 'ns':
```

Comment:
Once everything is protocol based, `int()` does not need knowledge of other types at all. This error check would be part of `NPDatetime.__int__`

Reply:
Removed it from the implementation.